### ISSUE TITLE
Manual linebreaks break smart quotes

### ISSUE THREAD

**Issue by Pegacraft** [typst/typst#1106]
### Description

When using a closing quote as the last part of a sentence, it doesn't get displayed properly

```
"Test 1"\
"Test 2"

"Test 1" \
"Test 2"

#set text(lang: "fr")
"Test 1"\
"Test 2"

```
Results in an output looking like this:
![image](https://user-images.githubusercontent.com/43176017/236304741-c8cdf4d4-9911-4d2f-bb66-9bad4e509604.png)

But expected would be:
![image](https://user-images.githubusercontent.com/43176017/236305029-7d9bf079-d723-49f0-a52a-1289f6b85f1b.png)

### Operating system

Web app

### Typst version

- [X] I am using the latest version of Typst

**Comment by andrewragy** [comment]
Fixed in 622cef8?

**Comment by laurmaedje** [comment]
> Fixed in https://github.com/typst/typst/commit/622cef8e0052532e9b71f7a2484e6637095b9b79?

Indeed.
